packages/components/src/mobile/html-text-input/index.native.js

```diff
@@ -24,11 +25,9 @@ export class HTMLTextInput extends Component {
 
 		this.edit = this.edit.bind( this );
 		this.stopEditing = this.stopEditing.bind( this );
+		addAction( 'native-editor.persist-html', 'core/editor', this.stopEditing );
```

Comment:
I wasn't familar with actions before looking at this PR. Would you mind briefly sharing when it makes sense to use them?

Reply:
Sorry I did not see your review, guess I was not very attentive of that PR!

I think it makes sense to use them when we have to listen to changes that are outside of or control.

In WordPress (web) hooks (actions and filters) are used all the time by plugins and widgets to modify the default behavior of the CMS (add scripts, listen to events etc).

In our case (gutenberg mobile) we control the whole codebase so we should be avoiding them in general. However we kind of have a code smell here in the way we implemented the parent <-> RN communication. Here the parent app decides when it wants to save the post content, but we actually have 2 editors which are disconnected from each others:
- the main block editor which is connected to the store
- the HTML editor which uses its own state

For us to be able to catch when the parent app requires the HTML content we need to subscribe to an event using the bridge (`subscribeParentGetHtml`), this is what's done in `NativeEditorProvider` which is btw always mounted and a parent of both the block editor and the html one. So whenever `serializeToNativeAction` is called we currently look at the store to serialize the blocks and send back the HTML, however since the current state of the HTML block editor is not saved in the store we need to "force a save" first and that's what the hook is for, to inform that the parent is requesting the HTML and that it's now the time for every component to do their things because the app might just quit.

An alternative would be to subscribe to the bridge inside `HTMLTextInput` but then you need to be absolutely sure `serializeToNativeAction` in `NativeEditorProvider` isn't going to be run just after your callback and overwrite what you just sent. So we need to resolve this somewhere. I haven't found a better way yet without modifying the bridge but I'm open to suggestions if you have other ideas

Reply:
Alternative ways I can think of:
- 1. Implement an autosave mechanism on the HTML editor (so fork the autosave component as it's not the case on web atm)
- 2. Emit 2 events from parent: 1 to tell the app that components need to cleanup/persist data (as it might quit) and another one to get the html from the app (`subscribeParentGetHtml`)